src/Network/Email/Email.php

```diff
@@ -1885,4 +1886,56 @@ protected function _getContentTypeCharset()
         }
         return strtoupper($this->charset);
     }
+
+    /**
+     * Serializes the email object to a value that can be natively serialized and re-used
+     * to clone this email instance.
+     *
+     * @return array Serializable array of configuration properties.
+     */
+    public function jsonSerialize()
+    {
+        $properties = [
+            'to', 'from', 'replyTo', 'cc', 'bcc', 'subject', 'returnPath', 'readReceipt',
+            'template', 'layout', 'viewRender', 'viewVars', 'theme', 'helpers', 'emailFormat', 'transport',
+            'attachments', 'domain', 'messageId', 'headers', 'charset', 'headerCharset',
+        ];
+
+        $array = [];
+
+        foreach ($properties as $property) {
+            $protected = '_' . $property;
```

Comment:
Totally right - I was first using the methods to retrieve the values and so I didn't have underscores - and then switched to properties since it would be faster to directly get/set properties instead of going through the methods (was doing methods because of my previous implementation in the custom type i showed you). Will fix that.